### DISCUSSION TITLE
Compile error for the board Generic ESP8266 Module.

### DISCUSSION THREAD

**Discussion by c7v** [arendst/Tasmota#17462] [Q&A] [answered]
In file included from D:\Tasmota-pre-release-12.3\tasmota\tasmota.ino:31:0:
D:\Tasmota-pre-release-12.3\tasmota\include\tasmota_globals.h:69:87: fatal error: include/tasmota_configurations.h: No such file or directory
 #include "include/tasmota_configurations.h"            // Preconfigured configurations
                                                                                       ^
compilation terminated.
exit status 1
Ошибка компиляции для платы Generic ESP8266 Module.

**Comment by Jason2866** [comment]
Using `.` in path can be problematic. Please try a path and folder without special characters.

**Reply by barbudor** [reply]
@c7v 
It looks like you have downloaded from Github a internal branch that is not supported

Please use ONLY branch `master` or branch `development`
There is no guarantee that other branch are working

**Comment by c7v** [comment]
Removed all characters from the path, downloaded the code from the master branch.

![image](https://user-images.githubusercontent.com/50216354/208826733-6ec0cad5-3b43-469b-ac03-869e9de15da6.png)

```
Arduino: 1.8.19 (Windows 10), Плата:"Generic ESP8266 Module, 80 MHz, Flash, Disabled (new can abort), All SSL ciphers (most compatible), dtr (aka nodemcu), 26 MHz, 40MHz, DOUT (compatible), 1MB (FS:64KB OTA:~470KB), 2, nonos-sdk 2.2.1+100 (190703), v2 Lower Memory, Disabled, None, Only Sketch, 115200"

In file included from D:\Tasmota\tasmota\tasmota.ino:31:0:

D:\Tasmota\tasmota\include\tasmota_globals.h:69:87: fatal error: include/tasmota_configurations.h: No such file or directory

 #include "include/tasmota_configurations.h"            // Preconfigured configurations

compilation terminated.
exit status 1
Ошибка компиляции для платы Generic ESP8266 Module.
```

**Reply by c7v** [reply]
There is an instruction on how to reflash through Arduino IDE?

**Reply by barbudor** [reply]
Are you using Arduino IDE?

That will not work. This is not supported. You need to use PlatformIO

The simplest method, if you don't have Visual-Studio-Code and PlatformIO already installed is to use Jason's portable environment. Just a zip to download and unzip. Then from VSCode you can pull the code from the repository and compile with a click
See instructions here: https://tasmota.github.io/docs/Visual-Studio-Code/#compiling-tools

**Reply by c7v** [reply]
> Are you using Arduino IDE

Yes 🤣

Thanks for the tip. I thought it should work like this. I'll definitely test it tonight and report back.

**Reply by c7v** [reply]
![image](https://user-images.githubusercontent.com/50216354/208987301-052af133-481d-4c2a-b358-aff62fd92cd4.png)

Загружается долго, инструкция не понятная, что, куда, зачем? Может вы поможете и сделаете для меня прошивку формата bin?

**Reply by Jason2866** [reply]
This is an english speaking discussions forum. Please  translate your post. Thx!

**Reply by c7v** [reply]
> This is an english speaking discussions forum. Please translate your post. Thx!

It takes a long time to load, does not understand what, where, why? Can you help and make a bin format firmware for me?

![image](https://user-images.githubusercontent.com/50216354/208991353-6548c316-a172-4be0-bc58-b8675683e923.png)

------

I'm trying to upload firmware to ESP8266, it doesn't work.

![image](https://user-images.githubusercontent.com/50216354/208991179-4f110cc0-7614-4977-89b5-6ebfcd3c4360.png)

I only change the number of LEDs I need for WS2812, default quantity is 30, i need to make 100.

```
c:/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld.exe: .pio\build\tasmota-RU\libFrameworkArduino.a(core_esp8266_main.cpp.o):(.text._ZL12loop_wrapperv+0x4): undefined reference to `setup'
c:/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld.exe: .pio\build\tasmota-RU\libFrameworkArduino.a(core_esp8266_main.cpp.o):(.text._ZL12loop_wrapperv+0x8): undefined reference to `loop'
c:/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld.exe: .pio\build\tasmota-RU\libFrameworkArduino.a(core_esp8266_main.cpp.o): in function `loop_wrapper()':
core_esp8266_main.cpp:(.text._ZL12loop_wrapperv+0x21): undefined reference to `setup'
c:/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld.exe: core_esp8266_main.cpp:(.text._ZL12loop_wrapperv+0x2d): undefined reference to `loop'
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\tasmota-RU\firmware.elf] Error 1
================================================================= [FAILED] Took 58.71 seconds =================================================================
Environment    Status    Duration
-------------  --------  ------------
tasmota-RU     FAILED    00:00:58.709
============================================================ 1 failed, 0 succeeded in 00:00:58.709 ============================================================
```

**Reply by Jason2866** [reply]
If this is the only change you want, there is no need to compile. You can change when Tasmota is flashed.
The setting is persistent.

**Reply by c7v** [reply]
> If this is the only change you want, there is no need to compile. You can change when Tasmota is flashed. The setting is persistent.

I also thought so, there must be a method to change somewhere in the interface, but I came across a task that said that you need to change the default value inside the firmware. I have made changes, however, as we can see from the dialog, I cannot download the firmware. Tell me where to change the number of LEDs?

**Comment by barbudor** [comment, marked as answer]
Command [pixels](https://tasmota.github.io/docs/Commands/#pixels)

**Reply by c7v** [reply]
Thank you, I figured out where to execute this command!

![image](https://user-images.githubusercontent.com/50216354/208996143-df6623c0-5a14-4b31-a06c-a77c2a5faedc.png)

**Reply by barbudor** [reply]
Once Tasmota is flashed and connected to your network, open the web page and goto the console page
Then type the command in the console


I'm not sure the ESP Easy Flasher is one of the supported flasher.
I recommend using the web flasher at https://tasmota.github.io/install
Just choose the Com port and "Tasmota" and it takes care of everything including downloading the right binary
